admin-ui/src/component/reservations/requested/RequestedReservation.tsx

```diff
@@ -330,49 +299,80 @@ const TimeBlock = ({
   );
 };
 
-const RequestedReservation = (): JSX.Element | null => {
-  const { id } = useParams() as { id: string };
-  const [reservation, setReservation] = useState<ReservationType | undefined>(
-    undefined
-  );
-  const [workingMemo, setWorkingMemo] = useState<string>();
+const WorkingMemo = ({
+  initialValue,
+  reservationPk,
+  refetch,
+}: {
+  initialValue: string;
+  reservationPk: number;
+  refetch: () => void;
+}) => {
+  const [workingMemo, setWorkingMemo] = useState<string>(initialValue);
   const { notifyError, notifySuccess } = useNotification();
-
   const { t } = useTranslation();
 
-  const { loading, refetch } = useQuery<Query, QueryReservationByPkArgs>(
-    RESERVATION_QUERY,
-    {
-      fetchPolicy: "no-cache",
-      variables: {
-        pk: Number(id),
-      },
-      onCompleted: ({ reservationByPk }) => {
-        if (reservationByPk) {
-          setReservation(reservationByPk);
-          setWorkingMemo(reservationByPk.workingMemo || "");
-        }
-      },
-      onError: () => {
-        notifyError(t("RequestedReservation.errorFetchingData"));
-      },
+  const [updateWorkingMemo] = useMutation<
+    Mutation,
+    ReservationWorkingMemoMutationInput
+  >(UPDATE_WORKING_MEMO, {
+    onCompleted: () => {
+      refetch();
+      notifySuccess(t("RequestedReservation.savedWorkingMemo"));
+    },
+    onError: () => {
+      notifyError(t("RequestedReservation.errorSavingWorkingMemo"));
+    },
+  });
+
+  const updateMemo = (memo: string) =>
+    updateWorkingMemo({
+      variables: { pk: reservationPk, workingMemo: memo },
+    });
+
+  const handleSave = async () => {
+    try {
+      await updateMemo(workingMemo);
+    } catch (ex) {
+      notifyError(t("RequestedReservation.errorSavingWorkingMemo"));
     }
-  );
-
-  const [updateWorkingMemo] = useMutation<Mutation>(UPDATE_WORKING_MEMO);
-
-  const updateMemo = (input: ReservationWorkingMemoMutationInput) =>
-    updateWorkingMemo({ variables: { input } });
+  };
 
-  const ref = useRef<HTMLDivElement>(null);
+  return (
+    <>
+      <TextArea
+        label={t("RequestedReservation.workingMemoLabel")}
+        id="workingMemo"
+        helperText={t("RequestedReservation.workingMemoHelperText")}
+        value={workingMemo}
```

Comment:
You might want to sanitize the input at some point.

Reply:
Not sure if sanitization would make a difference.
Afaik React doesn't allow setting html without dangerouslySetInnerHTML. 

Quick test if it's possible to execute code through the memo textarea:
```
</textarea><script>console.log('baz')</script>
```
On a HTML playground this executes the JS (so saving it in database and executing on another users browser is bad). Here it does not execute which is what I expected since React should not allow unescaped html without dangerouslySetInnerHTML.